lib/rules/no-alert.js

```diff
@@ -96,7 +96,7 @@ module.exports = {
         schema: [],
 
         messages: {
-            unexpected: "Unexpected {{name}}."
+            unexpected: "{{name}}() is considered to be obtrusive and should be replaced by a more appropriate custom UI implementation."
```

Comment:
I'm not a fan of this language. We try not to be overly judgmental with our default errors messages. The point of the rule is not that `alert` is obtrusive, but rather, it was likely left in as a mistake.

Reply:
The wording isn't mine. I pulled it verbatim from the documentation for this rule.

https://eslint.org/docs/rules/no-alert

Reply:
Ah funny, I can't remember how long ago that was written.

I still don't think this change makes sense, though.

Reply:
The problem with the existing message:

`"Unexpected alert."`

Is that it does not answer two fundamental questions for developers:

1. What is wrong?
2. How do I fix it?

The updated message is an attempt to answer both of those questions quickly and succinctly.

To assume that developers just "left it in accidently", assumes that they already know why it is a problem and how to fix it.

Reply:
The goal of the message is not to provide all of the context you suggest. That’s what the documentation is for. The name “no-alert” implies you will get a message saying that there is an alert, and because you have enabled the rule, presumably you know how to fix it.

While I understand your desire to put everything into the message, that’s just not what our approach has been. Especially in this rule, which is one of our first, I just don’t see a lot of value in trying to provide additional context that ESLint can’t possibly know vs. letting them go to the docs page to learn more. Most of our rules beginning with “no-“ have messages of this form.